src/test/java/com/KacFlor/ShopSpring/controller/CustomerControllerTest.java

```diff
@@ -0,0 +1,153 @@
+package com.KacFlor.ShopSpring.controller;
+
+import static org.mockito.BDDMockito.given;
+import static org.mockito.Mockito.*;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.*;
+import static org.springframework.test.web.servlet.result.MockMvcResultHandlers.print;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.*;
+
+import com.KacFlor.ShopSpring.controllersRequests.CustomerUpdateRequest;
+import com.KacFlor.ShopSpring.model.Customer;
+import com.KacFlor.ShopSpring.model.Role;
+import com.KacFlor.ShopSpring.model.User;
+import com.KacFlor.ShopSpring.service.CustomerService;
+import com.KacFlor.ShopSpring.service.JwtService;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.MediaType;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.security.test.context.support.WithMockUser;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.web.context.WebApplicationContext;
+
+import java.util.List;
+
+@SpringBootTest
+@AutoConfigureMockMvc
+@ExtendWith(MockitoExtension.class)
+public class CustomerControllerTest{
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    @MockBean
+    private CustomerService customerService;
+
+    @Autowired
+    private WebApplicationContext webApplicationContext;
+
+    @MockBean
+    private JwtService jwtService;
+
+    private User user1;
+
+    private User user2;
+
+    private User hAdmin;
+
+    private Customer customer = new Customer();
+
+    private Customer customer1 = new Customer();
+
+    private Customer customer2 = new Customer();
+
+    @BeforeEach
+    public void setup(){
+        PasswordEncoder encoder = new BCryptPasswordEncoder();
+        user1 = new User("Test1", "encodedPassword1", customer1, Role.USER);
+        user2 = new User("Test2", "encodedPassword2", customer2, Role.USER);
+        hAdmin = new User("HAdmin", encoder.encode("1234"), customer, Role.ADMIN);
+        customer1.setUser(user1);
+        customer2.setUser(user2);
+        customer.setUser(hAdmin);
+        customer1.setFirstName("Test1");
+        customer2.setFirstName("Test2");
+        customer.setFirstName("HAdmin");
```

Comment:
Why are these variables needed, where are they used?

Reply:
i use this for check content i get in the end of testGetCustomersList and testGetCustomer

Reply:
Ok I see, but declare them then inside test methods for each test. I know these can lead to code repetition but the test are more clean that way. These kind of global class variables do not look good, my first thought was that it is a complete spaghetti.

Reply:
done